src/bthread/task_group.cpp

```diff
@@ -156,11 +165,29 @@ bool TaskGroup::wait_task(bthread_t* tid) {
                 poll_start_ms = FLAGS_worker_polling_time_ms > 0 ? butil::cpuwide_time_ms() : 0;
                 continue;
             }
-            _pl->wait(_last_pl_state);
-            poll_start_ms = FLAGS_worker_polling_time_ms > 0 ? butil::cpuwide_time_ms() : 0;
-            if (update_ext_proc_) {
-                update_ext_proc_(1);
+            // bool woken_by_external = wait();
+            if (FLAGS_worker_cv_notify) {
+                LOG(INFO) << "group: " << group_id_ << "wait on cv";
+                bool woken_externally = wait();
```

Comment:
As double checking, you should check the "emptiness" of the cc shard before entering ``wait()``.